frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,175 @@
+
+    await openModal(user);
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('should close Dialog', async () => {
+    const user = userEvent.setup();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+    await openModal(user);
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(queriesMock.updateOptionList).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call doReloadPreview when closing Dialog', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved({
+      previewContextProps: { doReloadPreview },
+    });
+
+    await openModal(user);
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(doReloadPreview).toHaveBeenCalledTimes(1);
+    expect(queriesMock.updateOptionList).toHaveBeenCalledTimes(2);
+  });
+
+  it('should call updateOptionList when closing Dialog with correct parameters', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    await renderOptionListEditorAndWaitForSpinnerToBeRemoved({
+      previewContextProps: { doReloadPreview },
+    });
+    const expectedResultAfterEdit: Option[] = [
+      { value: 'test', label: 'label text', description: 'description', helpText: 'help text' },
+      { value: 2, label: 'label number', description: 'test', helpText: '' },
+      { value: true, label: 'label boolean', description: '', helpText: '' },
+    ];
+
+    await openModal(user);
+
+    const textBox = screen.getByRole('textbox', {
+      name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+    });
+    expect(textBox).toHaveTextContent('');
+    await user.type(textBox, 'test');
+
+    expect(
+      screen.getByRole('textbox', {
+        name: textMock('ux_editor.modal_properties_code_list_item_description', { number: 2 }),
+      }),
+    ).toHaveValue('test');
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(queriesMock.updateOptionList).toHaveBeenCalledTimes(3);
+    expect(queriesMock.updateOptionList).toHaveBeenCalledWith(
+      org,
+      app,
+      mockComponent.optionsId,
+      expectedResultAfterEdit,
+    );
+  });
+});
```

Comment:
I guess it is only nice with redundancy, but testing that the content is correct is testing the functionality of StudioCodeListEditor, which should be extensively tested in studio-components. 
But of course nice to test that the correct query is called when closing the modal!

Reply:
And the query should only be called once, or? Since we are not saving with debounce